zebra-network/src/meta_addr.rs

```diff
@@ -556,7 +574,8 @@ impl MetaAddrChange {
                 ..
             } => Some(*untrusted_last_seen),
             NewAlternate { .. } => None,
-            NewLocal { .. } => None,
+            // We know that our local listener is available
+            NewLocal { .. } => Some(DateTime32::now()),
```

Comment:
This was another bug: we didn't set the time on our local listener change, so sanitization was deleting it.